lib/instance.js

```diff
@@ -147,6 +147,65 @@ function instance(system) {
 		}
 	})
 
+	self.system.on('instance_http_req', function (req, res) {
+		let instance
+		res.set('Content-Type', 'text/plain')
+
+		for (const id in self.active) {
+			if (id === req.params.label || self.active[id].label === req.params.label) {
+				instance = self.active[id]
+				break
+			}
+		}
+
+		if (instance && typeof instance.handleHttpRequest === 'function') {
+			const requestData = {
+				baseUrl: req.baseUrl,
+				body: req.body,
+				headers: req.headers,
+				hostname: req.hostname,
+				ip: req.ip,
+				method: req.method,
+				originalUrl: req.originalUrl,
+				path: req.path,
+				query: req.query,
+			}
+
+			const httpTimeout = () => {
+				const delay = 30000
+				const defaultResponse = {
+					status: 500,
+					body: JSON.stringify({ status: 500, message: 'Internal server error' }),
```

Comment:
This might be better as a 502 Gateway timeout

Reply:
I went with a generic 500 as that'd cover whatever the error may be.

There's also the possibility for modules themselves to do their own errors and return more specific error codes/messages themselves as long as they do so before that default delay.

Reply:
Changed to 504 Gateway Timeout.

I do think in the future we may want to consider more specific error messages, to help indicate if it's on the side of Companion, or the Module. Generic errors should be fine for the time being though.